### PR TITLE
skip csi-addons pod restart for new namespaces

### PR DESCRIPTION
**Pull request by Nikhil-Ladha**
The csi-addons pod shouldn't be restarted when new namespaces starting with "openshift-" are created. We should only annotate the namespace and skip the csi-addons pod restart.

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2257259

### CODE REVIEW

File: controllers/namespace/namespace_controller.go [hunk 1]
```diff
@@ -23,6 +23,9 @@ import (
 // once is a pointer to the sync.Once type
 var once sync.Once
 
+// skipCSIAddonsPodRestart a bool type to check if the csi-Addons should be restarted
+var skipCSIAddonsPodRestart bool
```
Comment (iamniting) [inline controllers/namespace/namespace_controller.go:27]: Lets rename this variable to opposite
`triggerCSIAddonsPodRestart`
`performCSIAddonsPodRestart`
`executeCSIAddonsPodRestart`

**Review by iamniting — CHANGES_REQUESTED** [review]

File: controllers/namespace/namespace_controller.go (hunk 1 above)
Comment (Nikhil-Ladha) [reply controllers/namespace/namespace_controller.go]: Ak, updated to `performCSIAddonsPodRestart`

File: controllers/namespace/namespace_controller.go [hunk 2]
```diff
@@ -78,7 +81,7 @@ func (r *NamespaceReconciler) Reconcile(ctx context.Context, request reconcile.R
 	}
 
 	// Restart the csi-addons controller pod if any namespace was annotated
-	if updated {
+	if !performCSIAddonsPodRestart && updated {
```
Comment (iamniting) [inline controllers/namespace/namespace_controller.go:84]: not should be removed now.

**Review by iamniting — CHANGES_REQUESTED** [review]

File: controllers/namespace/namespace_controller.go [hunk 3]
```diff
@@ -23,6 +23,9 @@ import (
 // once is a pointer to the sync.Once type
 var once sync.Once
 
+// performCSIAddonsPodRestart a bool type to check if the csi-Addons pod should be restarted
+var performCSIAddonsPodRestart bool
```
Comment (Madhu-1) [inline controllers/namespace/namespace_controller.go:27]: as its global should we take lock. not required now but if we enable max-concurrent-reconciles we will get into problems

Comment (Nikhil-Ladha) [reply controllers/namespace/namespace_controller.go:27]: I think we are fine for now, we can update it later on when we enable max-concurrent-reconciles.

File: controllers/namespace/namespace_controller.go (hunk 2 above)
Comment (Nikhil-Ladha) [reply controllers/namespace/namespace_controller.go]: Done

File: controllers/namespace/namespace_controller.go [hunk 4]
```diff
@@ -192,9 +195,11 @@ func (r *NamespaceReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 	nameSpacePredicate := predicate.Funcs{
 		CreateFunc: func(e event.CreateEvent) bool {
+			performCSIAddonsPodRestart = true
```
[the remark this answers is not in the document]
Comment (Nikhil-Ladha) [reply controllers/namespace/namespace_controller.go]: Yeah, updated it now

**Comment by Nikhil-Ladha** [comment]
The openshift cluster installation is failing for the bundle-e2e test, seems like an issue with aws or so :/

**Review by Madhu-1 — COMMENTED** [review]
LGTM

**Review by iamniting — APPROVED** [review]

**Pull request merged**